src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/CodeActions/RazorCodeActionRunner.ts

```diff
@@ -0,0 +1,34 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { RazorLanguageServerClient } from '../RazorLanguageServerClient';
+import { RazorLogger } from '../RazorLogger';
+import { CodeActionResolutionResponse } from '../RPC/CodeActionResolutionResponse';
+import { convertWorkspaceEditFromSerializable } from '../RPC/SerializableWorkspaceEdit';
+
+export class RazorCodeActionRunner {
+
+    constructor(
+        private readonly serverClient: RazorLanguageServerClient,
+        private readonly logger: RazorLogger,
+    ) {}
+
+    public register() {
+        vscode.commands.registerCommand('razor/runCodeAction', (request: object) => this.runCodeAction(request), this);
+    }
+
+    private async runCodeAction(request: any): Promise<boolean> {
```

Comment:
👏 nice tiny shim

Reply:
Should request be `any`? I see below you acces its `Action` and `Data` properties. I'd imagine it'd be of type `CodeActionResolutionRequest` since you declare that in this PR too.